### PR TITLE
Add satelite resource assembly support

### PR DESCRIPTION
**Pull request by pakrym**
@davidfowl

### CODE REVIEW

File: src/Microsoft.DotNet.Tools.Compiler/Program.cs [hunk 1]
```diff
@@ -628,57 +633,109 @@ private static bool PrintSummary(List<DiagnosticMessage> diagnostics, Stopwatch
...
+            }
+
+            var name = ResourceManifestName.CreateManifestName(resourceName, rootNamespace);
+            return name;
+        }
+
+        private static bool GenerateResourceAssemblies(ProjectContext context, string outputPath, string configuration)
+        {
+            var project = context.ProjectFile;
+            var references = context.CreateExporter(configuration).GetDependencies().SelectMany(libraryExport => libraryExport.CompilationAssemblies);
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Compiler/Program.cs]: Why do this twice? Don't we already have the references?

File: src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            {
+                throw new ArgumentNullException(nameof(outputStream));
+            }
+
+            if (!sourceFiles.Any())
+            {
+                throw new InvalidOperationException("No source files specified");
+            }
+
+            List<ResourceDescription> resourceDescriptions = new List<ResourceDescription>();
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs]: Var

**Comment by pakrym** [comment]
:up: :date:

File: src/Microsoft.DotNet.Compiler.Common/CultureInfoCache.cs [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+    /// </summary>
+    internal static class CultureInfoCache
+    {
+        /// <summary>
+        /// This list of known cultures was generated by CultureInfoGenerator using .NET Framework 4.6 RC or later on
+        /// Microsoft Windows NT 6.2.9200.0.
+        /// As new versions of .NET Framework and Windows are released, this list should be regenerated to ensure it
+        /// contains the latest culture names.
+        /// </summary>
+        public static readonly HashSet<string> KnownCultureNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```
Comment (krwq) [inline src/Microsoft.DotNet.Compiler.Common/CultureInfoCache.cs]: bummer it's not somewhere in the System.Text - should you file an issue on dotnet/corefx?

File: src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            }
+
+            var compilationOptions = new CSharpCompilationOptions(outputKind: OutputKind.DynamicallyLinkedLibrary);
+            var compilation = CSharpCompilation.Create(asemblyName,
+                references: references.Select(reference => MetadataReference.CreateFromFile(reference)),
+                options: compilationOptions);
+
+            if (!string.IsNullOrEmpty(culture))
+            {
+                compilation = compilation.AddSyntaxTrees(new[]
```
Comment (anurse) [inline src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs]: Are we going to need to worry about making sure the assembly version of the resources dll matches the neutral assembly? Anyone know?

Comment (pakrym) [inline src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs]: Works fine on 451 as is

Comment (anurse) [inline src/Microsoft.DotNet.Tools.Resgen/ResourceAssemblyGenerator.cs]: :+1: sweet

**Pull request merged**